### PR TITLE
Port packaging scripts from Cirq

### LINKED ISSUE #483: Development releases
Currently, openfermion-cirq has to point at this github repository in order to use in-development features of openfermion during development of openfermion-cirq. This causes a bunch of problems w.r.t. packaging, since github dependencies aren't valid dependencies that can be uploaded to pypi or used by setup.py. I recently started making development releases for cirq ([scripts are here](https://github.com/quantumlib/Cirq/tree/master/dev_tools/packaging), example package [here](https://pypi.org/project/cirq/0.4.0.dev4220181112191000/)) and it immediately solved the analogous problem between cirq and openfermion-cirq. You just depend on `cirq~=0.4.0.dev` and it grabs the latest dev release of 0.4.0, and this is a valid version that works with pypi and setup.py.

We should get the same thing working in openfermion.

### PR DESCRIPTION
**Pull request by kevinsung**
Fixes #483

### CODE REVIEW

**Comment by babbush** [comment]
Can you explain what this PR does?

**Comment by kevinsung** [comment]
It adds scripts related to packaging and uploading to PyPI. For instance, you can cut a new release with a one-line command. You can also cut a dev release, or upload a release to the Test PyPI and verify that the release works as expected, which is good to do before doing a real release.

File: dev_tools/packaging/publish-dev-package.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 The Cirq Developers
```
Comment (Strilanc) [inline dev_tools/packaging/publish-dev-package.sh]: Change this to the openfermion copyright notice.

File: dev_tools/packaging/produce-package.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 The Cirq Developers
```
Comment (Strilanc) [inline dev_tools/packaging/produce-package.sh]: Change this to the openfermion copyright notice.

File: dev_tools/packaging/verify-published-package.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    # Check that code runs without dev deps.
+    echo Checking that code executes
+    "${tmp_dir}/${PYTHON_VERSION}/bin/python" -c "import openfermion; print(openfermion.FermionOperator((1, 1)))"
+    "${tmp_dir}/${PYTHON_VERSION}/bin/python" -c "import openfermion; print(openfermion.QubitOperator((1, 'X')))"
+
+    # Run tests.
+    echo Installing pytest requirements
+    if [ "${PYTHON_VERSION}" = "python2" ]; then
+        "${tmp_dir}/${PYTHON_VERSION}/bin/pip" install --quiet pytest mock
```
Comment (Strilanc) [inline dev_tools/packaging/verify-published-package.sh]: Are these requirements the same for openfermion as they are for cirq and openfermioncirq? You probably don't need `mock`.

Just unconditionally do `"${tmp_dir}/${PYTHON_VERSION}/bin/pip" install --quiet pytest mock` regardless of version

**Comment by kevinsung** [comment]
@Strilanc I made the changes but staled your review

**Comment by kevinsung** [comment]
... but I need this right now. Anyway, it's pretty clear that I just did what you said.

**Pull request merged**